src/classes/frDonor.cls

```diff
@@ -64,90 +66,134 @@ public class frDonor extends frModel implements frSyncable{
     public Boolean sync() {
         Boolean result = false;
         Map<String, Object> request = getRequestBody();
-        supporterContact = null;
-        String frId = getFunraiseId();
-        //try to find a donor that's already been integrated, use their funraise ID
-        List<Contact> contacts = (List<Contact>)Database.query('select Id, fr_ID__c, LastName from Contact where fr_ID__c = :frId');
-        if(contacts != null && contacts.size() > 0) {
-            supporterContact = contacts.get(0);
+        for (String key : request.keySet()) {
+            System.debug('Key: ' + key + ', Value: ' + request.get(key));
         }
+        String frId = getFunraiseId();
+    
+        SObject supporter = null; // Generic SObject for both Account and Contact
+        String sObjectType; // To track whether we're dealing with Account or Contact
         
+        Boolean isPersonAccount = frUtil.hasNPCobjects();
```

Comment:
for a bit more clarity on my end, can we rename this boolean to isNPC or something similar?  Just to make it clear that we are branching logic based on NPC/non-NPC environment

Reply:
Changed to isNPC!